sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -182,7 +182,12 @@ protected void registeredWithMesos() {
             activeTasks.addAll(decommissionedTasks);
         }
 
-        killUnneededTasks(stateStore, activeTasks);
+        if (schedulerConfig.useLegacyUnneededTaskKills()) {
+            //This case is turned off by default.
+            legacyKillUnneededTasks(stateStore, activeTasks, logger);
```

Comment:
Maybe add an expiration date here in a comment, like "Remove this on or after Dec 2018" (for example).